contracts/ics-escrow/src/contract.rs

```diff
@@ -0,0 +1,138 @@
+use crate::state::{ADMIN_ADDRESS, CHANNEL_ID};
+
+const CONTRACT_NAME: &str = "crates.io:ics-escrow";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    let admin_address = deps.api.addr_validate(&msg.admin_address)?;
+    ADMIN_ADDRESS.save(deps.storage, &admin_address)?;
+    CHANNEL_ID.save(deps.storage, &msg.channel_id)?;
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("admin_address", admin_address.to_string())
+        .add_attribute("channel_id", msg.channel_id))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Withdraw {
+            nft_address,
+            token_id,
+            receiver,
+        } => execute_withdraw(deps, env, info, nft_address, token_id, receiver),
+        ExecuteMsg::Burn {
+            nft_address,
+            token_ids,
+        } => execute_burn(deps.as_ref(), info, nft_address, token_ids),
+    }
+}
+
+fn execute_withdraw(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    nft_address: String,
+    token_id: String,
+    receiver: String,
+) -> Result<Response, ContractError> {
+    let admin_address = ADMIN_ADDRESS.load(deps.storage)?;
+    if info.sender != admin_address {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Validate receiver and class_uri
+    deps.api.addr_validate(&nft_address)?;
+    deps.api.addr_validate(&receiver)?;
+
+    let transfer_msg = cw721::Cw721ExecuteMsg::TransferNft {
+        recipient: receiver,
+        token_id,
+    };
+    let msg = WasmMsg::Execute {
+        contract_addr: nft_address,
+        msg: to_binary(&transfer_msg)?,
+        funds: vec![],
+    };
+    Ok(Response::new()
+        .add_attribute("action", "withdraw")
+        .add_message(msg))
+}
+
+fn execute_burn(
+    deps: Deps,
+    info: MessageInfo,
+    nft_address: String,
+    token_ids: Vec<String>,
+) -> Result<Response, ContractError> {
+    let admin_address = ADMIN_ADDRESS.load(deps.storage)?;
+    if info.sender != admin_address {
+        return Err(ContractError::Unauthorized {});
+    }
+    deps.api.addr_validate(&nft_address)?;
+
+    let messages = token_ids
+        .into_iter()
+        .map(|token_id: String| -> StdResult<WasmMsg> {
+            Ok(WasmMsg::Execute {
+                contract_addr: nft_address.clone(),
+                // This works despite the fact that we need to be
+                // compatible with the cw721 base spec (which does not
+                // have a burn method) everywhere else.
+                //
```

Comment:
TODO remove when we bump `cw721` to `v0.13.4`.

Reply:
Updated the wording and package. Thanks!